modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj

```diff
@@ -146,6 +154,90 @@
                table-schema->metabase-field-info
                set)})
 
+(def ^:private ^DateTimeFormatter bq-ts-str-formatter
+  "For some crazy reason, the TIMESTAMP String values returned from [[FieldValue]] look like `1577836800.0`, so create
+  a parser for that.  This is basically '<epoch seconds>.<millis>'."
+  (let [^DateTimeFormatterBuilder fb (doto (DateTimeFormatterBuilder.)
+                                       (.appendValue ChronoField/INSTANT_SECONDS 1 19 SignStyle/NEVER)
+                                       (.appendLiteral \.)
+                                       (.appendValue ChronoField/MILLI_OF_SECOND))]
+    (.toFormatter fb)))
+
+(defn- bq-ts-str->instant
+  "Parses the given `ts-str` (assumed to be a String representing a BigQuery TIMESTAMP value) to an [[Instant]].  See
+  [[bq-ts-str-formatter]] for more details."
+  [ts-str]
+  (let [parsed (.parse bq-ts-str-formatter ts-str)]
+    (Instant/from parsed)))
+
+;; https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#integer_type
+(def ^:private int-fingerprint-types
+  "Integral types for fingerprinting purposes (to be parsed via BigInteger)"
+  #{"INT64" "INT" "SMALLINT" "INTEGER" "BIGINT" "TINYINT" "BYTEINT"})
+
+;; https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#decimal_types
+;; https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#floating_point_types
+(def ^:private dec-fingerprint-types
+  "Decimal and floating point types for fingerprinting purposes (to be parsed via BigDecimal)"
+  #{"NUMERIC" "DECIMAL" "BIGNUMERIC" "BIGDEIMAL" "FLOAT" "FLOAT64"})
+
+(defn- fields->indexed-parser-fns
```

Comment:
Question. Why couldn't you reuse the parsing code we already wrote for BigQuery in https://github.com/metabase/metabase/blob/master/modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk/query_processor.clj#L69-L132 ?

It seems like a lot of this stuff is duplicated with the existing code. 

If there's some reason that stuff wouldn't work as-is, I'd put money on it being easier to make a few small tweaks to it than to create a bunch of new code that does the same thing